metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/indexing_master.py

```diff
@@ -150,12 +152,48 @@ def zeppelin_notebook_import(self, env):
         env.set_params(params)
 
         Logger.info(ambari_format('Searching for Zeppelin Notebooks in {metron_config_zeppelin_path}'))
+
+        # With Ambari 2.5+, Zeppelin server is enabled to work with Shiro authentication, which requires user/password
+        # for authentication (see https://zeppelin.apache.org/docs/0.6.0/security/shiroauthentication.html for details).
+        ses = requests.session()
```

Comment:
Would it make sense to factor out the authentication logic into a separate function named something like `zeppelin_authenticate`?   That is a fairly complicated bit in and out of itself.